### PR TITLE
dagster-dbt types

### PR DESCRIPTION
**Pull request by smackesey**
### Summary & Motivation

Fixes some type errors for dagster-dbt, fixes the resource decorator type sig, and adds dagster-dbt to buildkite checking.

### How I Tested These Changes

BK

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #7867** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7867" target="_blank">Graphite</a> 
    * **PR #7812** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7812" target="_blank">Graphite</a> 
      * **PR #7813** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7813" target="_blank">Graphite</a> 
        * **PR #7700** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7700" target="_blank">Graphite</a> 
        * **PR #7878** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7878" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/7878?utm_source=stack-comment).

File: python_modules/libraries/dagster-dbt/dagster_dbt/ops.py [hunk 1]
```diff
@@ -103,4 +107,4 @@ def dbt_docs_generate_op(context):
     (dbt_seed_op, "seed"),
     (dbt_docs_generate_op, "docs generate"),
 ]:
-    op.__doc__ = _get_doc(op.name, cmd)
+    op.__doc__ = _get_doc(_op.name, cmd)
```
Comment (OwenKephart) [inline python_modules/libraries/dagster-dbt/dagster_dbt/ops.py:110]: this looks like it's setting the dagster.op docstring. I think instead this should be `for dbt_op, cmd ...` and `dbt_op.__doc__ = _get_doc(dbt_op.name, cmd)`

**Review by OwenKephart — APPROVED** [review]
one comment, but feel free to merge after that

File: python_modules/libraries/dagster-dbt/dagster_dbt/ops.py (hunk 1 above)
Comment (smackesey) [reply python_modules/libraries/dagster-dbt/dagster_dbt/ops.py:110]: good catch

**Pull request merged**